sdk/core/azure-core-tracing-opencensus/azure/core/tracing/ext/opencensus_span/__init__.py

```diff
@@ -61,7 +61,38 @@ def __init__(self, span=None, name="span", **kwargs):
         :paramtype links: list[~azure.core.tracing.Link]
         """
         tracer = self.get_current_tracer()
+        value = kwargs.pop('kind', None)
+        kind = (
+            OpenCensusSpanKind.CLIENT if value == SpanKind.CLIENT else
+            OpenCensusSpanKind.CLIENT if value == SpanKind.PRODUCER else # No producer in opencensus
+            OpenCensusSpanKind.SERVER if value == SpanKind.SERVER else
+            OpenCensusSpanKind.CLIENT if value == SpanKind.CONSUMER else # No consumer in opencensus
+            OpenCensusSpanKind.UNSPECIFIED if value == SpanKind.INTERNAL else # No internal in opencensus
+            OpenCensusSpanKind.UNSPECIFIED if value == SpanKind.UNSPECIFIED else
+            None
+        ) # type: SpanKind
+        if value and kind is None:
+            raise ValueError("Kind {} is not supported in OpenCensus".format(value))
+
+        links = kwargs.pop('links', None)
         self._span_instance = span or tracer.start_span(name=name, **kwargs)
+        if kind is not None:
+            self._span_instance.span_kind = kind
+
+        if links:
```

Comment:
Missed links= kwargs.pop('links', None). :)

Reply:
i have it in line 77